picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -664,13 +668,24 @@ void shutdown() {
     private ExecutorService service;
     private Cache cache;
     private Listener listener;
-    private RequestTransformer transformer;
-    private List<RequestHandler> requestHandlers;
+    private RequestTransformer requestTransformer;
+    private List<RequestHandler> requestHandlers = new ArrayList<>();
     private Bitmap.Config defaultBitmapConfig;
-
     private boolean indicatorsEnabled;
     private boolean loggingEnabled;
 
+    /** Start building a new {@link Picasso} instance. */
+    Builder(@NonNull Picasso picasso) {
+      context = picasso.context;
+      cache = picasso.cache;
+      listener = picasso.listener;
+      requestTransformer = picasso.requestTransformer;
+      requestHandlers.addAll(picasso.requestHandlers);
+      defaultBitmapConfig = picasso.defaultBitmapConfig;
+      indicatorsEnabled = picasso.indicatorsEnabled;
+      loggingEnabled = picasso.loggingEnabled;
+    }
```

Comment:
i need the OkHttpClient to be shared.

Reply:
I will look into exposing `OkHttpClient` since it initiated in `OkHttp3Downloader `.

Reply:
Maybe we can get this in as a first step? I see you created another issue to remove `Downloader`.